### PR TITLE
implement RadixSort

### PR DESCRIPTION
**Pull request by EAlexa**
Let's give this another try with RadixSort

### CODE REVIEW

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java [hunk 1]
```diff
@@ -1,8 +1,51 @@
-/** NOTE: This file is still under development */
 package com.williamfiset.algorithms.sorting;
 
 public class RadixSort {
```
Comment (williamfiset) [inline src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:3]: Can you add a description of the algorithm, how it works, what type of data you want to sort with it and some time complexity information?

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.williamfiset.algorithms.sorting;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.Test;
+
+public class RadixSortTest {
+  @Test
+  public void testGetMax() {
```
Comment (williamfiset) [inline src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:10]: Can you add more test coverage? Can you also add some randomized tests? E.g:

/src/main/java/com/williamfiset/algorithms/sorting/SelectionSort.java@master#L45

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java (hunk 1 above)
Comment (EAlexa) [reply src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:8]: Done :-)

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java (hunk 2 above)
Comment (EAlexa) [reply src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:11]: Random Test added

File: src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java (hunk 1 above)
Comment (williamfiset) [reply src/main/java/com/williamfiset/algorithms/sorting/RadixSort.java:8]: Thanks!

File: src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    assertTrue(2 == RadixSort.calculateNumberOfDigits(19));
+  }
+
+  @Test
+  public void testRadixSort() {
+    // Random Test
+    Random random = new Random();
+    int[] array2 = new int[50];
+    for (int i = 0; i < 50; i++) {
+      array2[i] = random.nextInt(50) + 1;
```
[the remark this answers is not in the document]
Comment (williamfiset) [reply src/test/java/com/williamfiset/algorithms/sorting/RadixSortTest.java:28]: nvm it's not

**Comment by williamfiset** [comment]
Added some small modifications, testing and patched a small bug in:  4bf2b73148ad3a3523d44a6939c90055f09085b8

**Pull request merged**